pkg/controller/core/workload_controller.go

```diff
@@ -231,18 +295,19 @@ func (r *WorkloadReconciler) Update(e event.UpdateEvent) bool {
 		if !r.cache.AddOrUpdateWorkload(wlCopy) {
 			log.V(2).Info("ClusterQueue for workload didn't exist; ignored for now")
 		}
-
-	case prevStatus == admitted && status == pending:
+	case prevStatus == admitted && status == cancellingAdmission:
```

Comment:
what about the case where preStatus=cancellingAdmission && status=admitted; if not possible, please document why.

Reply:
Added a note in the comment for the `admitted` to `cancellingAdmission`